ksqldb-api-client/src/test/java/io/confluent/ksql/api/client/integration/PushV2ClientContinueIntegrationTest.java

```diff
@@ -380,11 +383,31 @@ private static void addObjectToKsqlArray(final KsqlArray array, final Object val
       array.add(SqlTimeTypes.formatDate((Date) value));
     } else if (value instanceof Time) {
       array.add(SqlTimeTypes.formatTime((Time) value));
+    } else if (value instanceof byte[]) {
+      array.add(serializeVertX3CompatibleByte((byte[]) value));
     } else {
       array.add(value);
     }
   }
 
+  /**
+   * VertX 4 changed to serialize using Base64 without padding but our server
+   * still encodes the data with padding - this uses the same serializer that
+   * the server uses to serialize the data into a string
+   *
+   * @see <a href=https://github.com/eclipse-vertx/vert.x/pull/3197>vertx#3197</a>
+   */
+  private static String serializeVertX3CompatibleByte(final byte[] bytes) {
```

Comment:
Can we refactor this to a common place?  (Not a blocking question.)

Reply:
Does `-Dvertx.json.base64=legacy` help us at any point?  (I'm blindly cribbing from the docs...)

Reply:
> Can we refactor this to a common place? (Not a blocking question.)

yeah, but we can do that in a different PR 😆  this one was already getting a bit big

> Does `-Dvertx.json.base64=legacy` help us at any point? (I'm blindly cribbing from the docs...)

I tried using that but it's a really crappy workaround that they provide TBH, for one there's no way to do that programmatically because the choice is done in the static class loader so there's no guarantee that the class loader will run my `System.setProperty` call before their static class is loaded.

Then if we want to add it in the runtime configurations, there's no way to do that by default in IntelliJ (it won't respect the maven `pom` runtime parameters because it uses it's own runtime configuration) so even if I could get the test to pass in maven (which it did) you have to specify it manually when you run this test (which works, but is a bad experience)

There's also no guarantees that people will be running whatever scripts we choose when running their clients, so ultimately I think relying on that behavior isn't that straightforward.

Reply:
Yeah, feel free to push that change to another PR (or drop it; can't be DRY every day).

And thanks for trying that System property!  It does sound a little finicky.